interview-questions.md

```diff
@@ -9,35 +9,77 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. As a developer, I am creating a Rails application with a model called Cohort and a model called Student. The cohort model has_many students, but OOPS! I forgot to add the foreign key. How can I fix this mistake? What is the name of the foreign key? Would the foreign key be on the Cohort model or the Student model?
 
 Your answer:
+You will need to create a migration in the console. When the migration file is created, open it and add the id of the cohort model to the student model.
+the foreign key is the cohort id which is on the student model.
 
 Researched answer:
+Assuming students belongs_to cohort, you will need to generate a migration to add an additional column to the Student model. In the migration file you will
+need to use the add_column method. The code will look something like this: add_column :students, :cohort_id, :integer. What this is doing is adding 
+"cohort_id" (which is also the foreign key) column to the "students" table and the datatype is "integer". After you migrate the database students will now have that column. Since cohort has_many students, the foreign key on the student should match the primary key of the cohort.
+
+https://guides.rubyonrails.org/active_record_migrations.html
 
 2. Which RESTful routes must always be passed params? Why?
 
 Your answer:
+There are three RESTful routes that must be passed which are POST, PATCH and DELETE because they all require an id of the element.
 
 Researched answer:
+The RESTful routes that must be passed as params PUT, PATCH, and DELETE. Params are needed because each of these verbs require the ID of the 
```

Comment:
Keep in mind RESTful routes vs HTTP verbs. RESTful routes are conventions for defining URLs and HTTP methods used to access resources in a RESTful app. With this in mind, we can see that the routes that need params are show, edit, update, destroy. Confusing question, I know